Source/Core/Core/PowerPC/Jit64Common/BlockCache.h

```diff
@@ -13,6 +15,11 @@ class JitBlockCache : public JitBaseBlockCache
 public:
   explicit JitBlockCache(JitBase& jit);
 
+  void DestroyBlock(JitBlock& block) override;
+
+  std::vector<std::pair<u8*, u8*>> m_ranges_to_free_on_next_codegen_near;
```

Comment:
These should be private and instead have a class interface built around these.

Reply:
Before I change this, I'm looking at this again I'm not 100% sure when to exactly clear those vectors. I would have just put the clear into `JitBlockCache::Clear()` but it looks like that can be called without the Jit64's `Clear()` being called, which could desync what the Jit64 thinks is free versus what actually is. Am I missing something here? Does it ever make sense to clear the block cache without clearing the entire Jit?

Reply:
Wait, I think I'm thinking about this in the wrong direction. A block cache clear should actually *add* all of the destroyed blocks to the ranges-to-free vectors, and then the next Jit() will pick that up correctly. I think that works, let's see...